assets/js/search.js

```diff
@@ -2,91 +2,130 @@ $ = jQuery;
+	$( '.multiple-search-result' ).off( 'click', '.search-result-item-image, .search-result-item-content' ).on( 'click', '.search-result-item-image, .search-result-item-content', function() {
 		window.location.href = $( '.search-result-item-headline', $( this ).parent() ).attr( 'href' );
-    });
-
-    // Clear single selected filter.
-    $( '.activefilter-tag' ).off( 'click' ).on( 'click', function() {
-        $( '.custom-control-input[value=' + $( this ).data( 'id' ) + ']' ).prop('checked', false );
-        $search_form.submit();
-    });
-
-    // Clear all selected filters.
-    $( '.clearall' ).off( 'click' ).on( 'click', function() {
-        $( 'input[name^="f["]' ).prop( 'checked', false );
-        $search_form.submit();
-    });
-
-    // Add click event for load more button in blocks.
-    $load_more_button.off( 'click' ).on( 'click', function() {
-        var $row = $( '.row-hidden', $load_more_button.closest( '.container' ) );
-
-        if ( 1 === $row.size() ) {
-            $load_more_button.closest( '.load-more-button-div' ).hide( 'fast' );
-        }
-        $row.first().show( 'fast' ).removeClass( 'row-hidden' );
-    });
-
-    // Reveal more results just by scrolling down the first 2 times.
-    $( window ).scroll(function() {
-        if ( $load_more_button.length > 0 ) {
-            var element_top = $load_more_button.offset().top,
-                element_height = $load_more_button.outerHeight(),
-                window_height = $(window).height(),
-                window_scroll = $(this).scrollTop();
-
-            if (window_scroll > ( element_top + element_height - window_height )) {
-                load_more_count++;
-                if (load_more_count <= 2) {
-                    $load_more_button.click();
-                }
-                return false;
-            }
-        }
-    });
+	});
+
+	// Clear single selected filter.
+	$( '.activefilter-tag' ).off( 'click' ).on( 'click', function() {
+		$( '.custom-control-input[value=' + $( this ).data( 'id' ) + ']' ).prop('checked', false );
+		$search_form.submit();
+	});
+
+	// Clear all selected filters.
+	$( '.clearall' ).off( 'click' ).on( 'click', function() {
+		$( 'input[name^="f["]' ).prop( 'checked', false );
+		$search_form.submit();
+	});
+
+	// Add click event for load more button in blocks.
+	$load_more_button.off( 'click' ).on( 'click', function() {
+
+		// If this button has this class then Lazy-loading is enabled.
+		if ( $(this).hasClass( 'btn-load-more-async' ) ) {
+			var next_page = $(this).data( 'current_page' ) + 1;
+			$(this).data( 'current_page', next_page );
+
+			$.ajax({
+				url: ajaxurl,
+				type: 'GET',
+				data: {
+					action:          'get_paged_posts',
+					'search-action': 'get_paged_posts',
+					'paged':         next_page,
+					'query-string':  decodeURIComponent( location.search ).substr( 1 )		// Ignore the ? in the search url (first char).
+				},
+				dataType: 'html'
+			})
+			.done(function ( response ) {
```

Comment:
Done callback will run for all cases (success, error). Maybe it is better to separate the actions taken in each case (successful response or error occured during the request ) using success and error callback functions.
```javascript
$.ajax({

	success: function (response) {
	},
	error: function (jqXHR, textStatus, errorThrown) {
	}
}).always(function () {

});
```

Reply:
Nop, 'done()' executes on 'success()' and 'always()' executes always.
The new recommended way is to chain call done(), fail(), always() (so I guess I should add .fail() and .always() calls).
Moreover, these methods also implement the Promise interface, giving them all the properties, methods, and behavior of a Promise giving more flexibility in the future.